### PR TITLE
update bin2tif for reprocessing 2018

### PR DESCRIPTION
**Pull request by max-zilla**

### CODE REVIEW

File: bin2tif/Dockerfile [hunk 1]
```diff
@@ -26,4 +20,7 @@ CMD ["extractor"]
 ENV RABBITMQ_EXCHANGE="terra" \
     RABBITMQ_VHOST="%2F" \
     RABBITMQ_QUEUE="terra.stereo-rgb.bin2tif" \
-    MAIN_SCRIPT="terra_bin2tif.py"
+    MOUNTED_PATHS='{"home/clowder/sites":"/home/extractor/sites"}' \
```
Comment (max-zilla) [inline bin2tif/Dockerfile]: don't need MOUNTED_PATHS here because we have it in the admin repo YAML file.

File: bin2tif/extractor_info.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "@context": "http://clowder.ncsa.illinois.edu/contexts/extractors.jsonld",
   "name": "terra.stereo-rgb.bin2tif",
-  "version": "1.3",
+  "version": "2.1",
```
Comment (max-zilla) [inline bin2tif/extractor_info.json]: 1.4

File: bin2tif/terra_bin2tif.py [hunk 3]
```diff
@@ -80,81 +83,169 @@ def process_message(self, connector, host, secret_key, resource, parameters):
...
-            # Rename output.tif after creation to avoid long path errors
-            create_geotiff(left_image, left_gps_bounds, out_tmp_tiff, None, False, self.extractor_info, metadata)
             # TODO: we're moving zero byte files
-            shutil.move(out_tmp_tiff, left_tiff)
-            if left_tiff not in resource['local_paths']:
-                fileid = upload_to_dataset(connector, host, self.clowder_user, self.clowder_pass, target_dsid, left_tiff)
-                uploaded_file_ids.append(host + ("" if host.endswith("/") else "/") + "files/" + fileid)
+            if os.path.getsize(out_tmp_tiff_left) > 0:
+                shutil.move(out_tmp_tiff_left, left_tiff)
+                if left_tiff not in resource['local_paths']:
```
Comment (max-zilla) [inline bin2tif/terra_bin2tif.py]: here let's use the found_in_dataset() model:
https://github.com/terraref/extractors-multispectral/pull/32/files#diff-a186da4a4378b6d0a96e0c8abd3bc92bR158

File: bin2tif/terra_bin2tif.py [hunk 4]
```diff
@@ -80,81 +83,169 @@ def process_message(self, connector, host, secret_key, resource, parameters):
...
-            if right_tiff not in resource['local_paths']:
-                fileid = upload_to_dataset(connector, host, self.clowder_user, self.clowder_pass, target_dsid,right_tiff)
-                uploaded_file_ids.append(host + ("" if host.endswith("/") else "/") + "files/" + fileid)
+            right_image = terraref.stereo_rgb.process_raw(left_shape, img_right, None)
+            out_tmp_tiff_right = os.path.join(tempfile.gettempdir(), resource['id'].encode('utf8'))
+            # Rename output.tif after creation to avoid long path errors
+            create_geotiff(right_image, gps_bounds, out_tmp_tiff_right, None, True, self.extractor_info, terra_md_full)
+            if os.path.getsize(out_tmp_tiff_right) > 0:
+                shutil.move(out_tmp_tiff_right, right_tiff)
+                if left_tiff not in resource['local_paths']:
```
Comment (max-zilla) [inline bin2tif/terra_bin2tif.py]: same here

File: bin2tif/terra_bin2tif.py [hunk 5]
```diff
@@ -80,81 +83,169 @@ def process_message(self, connector, host, secret_key, resource, parameters):
...
+                self.bytes += os.path.getsize(left_tiff)
             else:
-                self.log_info(resource, "file found in dataset already; not re-uploading")
-            self.created += 1
-            self.bytes += os.path.getsize(right_tiff)
+                self.log_info("Zero bytes file generated")
+
+
+        # Plot dir is the day under Level_1_Plots/ir_geotiff/day
+        # TODO
```
Comment (max-zilla) [inline bin2tif/terra_bin2tif.py]: this whole block should be the new plot clipping extractor we need to write. we will submit the files we've just uploaded to clowder to it during THIS extractor, and then that extractor will do this clipping bit

File: bin2tif/Dockerfile (hunk 1 above)
Comment (tcnichol) [reply bin2tif/Dockerfile]: removed

File: bin2tif/terra_bin2tif.py (hunk 5 above)
Comment (tcnichol) [reply bin2tif/terra_bin2tif.py]: fixed

File: bin2tif/terra_bin2tif.py (hunk 3 above)
Comment (tcnichol) [reply bin2tif/terra_bin2tif.py]: fixed

File: bin2tif/terra_bin2tif.py (hunk 4 above)
Comment (tcnichol) [reply bin2tif/terra_bin2tif.py]: fixed

**Pull request merged**